jwt/api.py

```diff
@@ -110,14 +129,16 @@ def encode(self, payload, key, algorithm='HS256', headers=None, json_encoder=Non
 
         return b'.'.join(segments)
 
-    def decode(self, jwt, key='', verify=True, algorithms=None, **kwargs):
+    def decode(self, jwt, key='', verify=True, algorithms=None, options=None, **kwargs):
         payload, signing_input, header, signature = self._load(jwt)
 
         if verify:
-            self._verify_signature(payload, signing_input, header, signature,
-                                   key, algorithms)
+            merged_options = self._merge_options(override_options=options)
+            if merged_options.get('verify_signature'):
+                self._verify_signature(payload, signing_input, header, signature,
+                                       key, algorithms)
```

Comment:
We can always iterate on this further. I think this design is a big improvement over what we have currently. I'm open to PRs suggesting further enhancements before we do a release.

The nice thing about code is that we can change it (until a release; then we have to consider API changes and how they will affect consumers)